src/Controls/src/Build.Tasks/SetPropertiesVisitor.cs

```diff
@@ -291,11 +291,19 @@ internal static string GetContentProperty(XamlCache cache, TypeReference typeRef
 				if (!acceptEmptyServiceProvider && requireServiceAttribute == null)
 					context.LoggingHelper.LogWarningOrError(BuildExceptionCode.UnattributedMarkupType, context.XamlFilePath, node.LineNumber, node.LinePosition, 0, 0, vardefref.VariableDefinition.VariableType);
 
-				if (vardefref.VariableDefinition.VariableType.FullName == "Microsoft.Maui.Controls.Xaml.BindingExtension"
-					&& bpRef != null //do not compile bindings if we're not gonna SetBinding
-					)
-					foreach (var instruction in CompileBindingPath(node, context, vardefref.VariableDefinition))
-						yield return instruction;
+				if (bpRef is not null) // do not compile bindings if we're not gonna SetBinding
```

Comment:
I hope we can remove this condition in https://github.com/dotnet/maui/pull/22023